test/ziggurat/streams_test.clj

```diff
@@ -102,6 +103,67 @@
     (is (= times @message-received-count))))
 
+(deftest stop-stream-test
+  (let [message-received-count (atom 0)
+        mapped-fn              (get-mapped-fn message-received-count)
+        times                  6
+        kvs                    (repeat times message-key-value)
+        handler-fn             (default-middleware/protobuf->hash mapped-fn proto-class :default)
+        _                      (mount/start)
+        streams                (start-streams {:default {:handler-fn handler-fn}}
+                                              (-> (ziggurat-config)
+                                                  (assoc-in [:stream-router :default :application-id] (rand-application-id))
+                                                  (assoc-in [:stream-router :default :changelog-topic-replication-factor] changelog-topic-replication-factor)))]
+    (Thread/sleep 10000)                                    ;;waiting for streams to start
+    (IntegrationTestUtils/produceKeyValuesSynchronously (get-in (ziggurat-config) [:stream-router :default :origin-topic])
+                                                        kvs
+                                                        (props)
+                                                        (MockTime.))
+    (Thread/sleep 10000)                                     ;;wating for streams to consume messages
+    (stop-stream :default)
+    (is (= times @message-received-count))))
+
+(deftest stop-duplicate-stream-test
+  (let [message-received-count (atom 0)
+        mapped-fn              (get-mapped-fn message-received-count)
+        times                  6
+        kvs                    (repeat times message-key-value)
+        handler-fn             (default-middleware/protobuf->hash mapped-fn proto-class :default)
+        _                      (mount/start)
+        streams                (start-streams {:default {:handler-fn handler-fn}}
+                                              (-> (ziggurat-config)
+                                                  (assoc-in [:stream-router :default :application-id] (rand-application-id))
+                                                  (assoc-in [:stream-router :default :changelog-topic-replication-factor] changelog-topic-replication-factor)))]
+    (Thread/sleep 10000)                                    ;;waiting for streams to start
+    (IntegrationTestUtils/produceKeyValuesSynchronously (get-in (ziggurat-config) [:stream-router :default :origin-topic])
+                                                        kvs
+                                                        (props)
+                                                        (MockTime.))
+    (Thread/sleep 10000)                                     ;;wating for streams to consume messages
+    (stop-stream :default)
+    (stop-stream :default)                                   ;;attempting to close same stream again
+    (is (= times @message-received-count))))
```

Comment:
Same as for line number 124.

Also, in all these tests why are we publishing and consuming from streams?

Reply:
Updated same as above.

Yes, we don't need to consume from the streams. Removed that. 
We do need to publish them because otherwise the component life cycle of `defstate stream` wouldn't start and when we make the actual call to close the stream, the `stream` component would have KafkaStreams which are in `NOT_STARTED` state. Publishing the streams allows us to test the actual stopping of the stream.